stacks/spica/src/storage/components/storage/storage.component.ts

```diff
@@ -52,16 +55,22 @@ export class StorageComponent implements ControlValueAccessor {
 
     if (files.length) {
       this.blob = files.item(0);
-      this.progress$ = this.storage.insertMany(files).pipe(
-        map(event => {
-          if (event.type === HttpEventType.UploadProgress) {
-            return Math.round((100 * event.loaded) / event.total);
-          } else if (event.type === HttpEventType.Response) {
-            this.value = event.body[0].url;
-            this.onChangeFn(this.value);
-            this.progress$ = undefined;
-          }
-        }),
+
+      this.progress$ = this.rootDir.find("root").pipe(
+        switchMap(r => (r ? of(r) : this.rootDir.add("root").toPromise())),
```

Comment:
No need to use "switchMap" operator cause you are not waiting for anything.
Try using it as I wrote below.

  ```
   this.progress$ = this.rootDir.find("root").pipe(
        tap((r)=>!r &&  this.rootDir.add("root").toPromise())
        switchMap(() =>
```